File: laundry-reservation/typescript/tests/RecordingEmailNotifier.ts
import type { EmailNotifier } from '../src/EmailNotifier.js';

export interface EmailMessage {
  to: string;
  subject: string;
  body: string;
}

export class RecordingEmailNotifier implements EmailNotifier {
  readonly sent: EmailMessage[] = [];

  send(to: string, subject: string, body: string): void {
    this.sent.push({ to, subject, body });
  }
}

File: laundry-reservation/python/tests/recording_email_notifier.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class EmailMessage:
    to: str
    subject: str
    body: str


class RecordingEmailNotifier:
    def __init__(self) -> None:
        self.sent: list[EmailMessage] = []

    def send(self, to: str, subject: str, body: str) -> None:
        self.sent.append(EmailMessage(to, subject, body))
